src/main/scala/cognite/spark/v1/RawTableRelation.scala

```diff
@@ -220,19 +210,64 @@ object RawTableRelation {
   private def lastUpdatedTimeColumns(schema: StructType): Array[String] =
     schema.fieldNames.filter(lastUpdatedTimeColumnPattern.findFirstIn(_).isDefined)
 
-  def rowsToRawItems(nonKeyColumnNames: Seq[String], rows: Seq[Row], mapper: ObjectMapper): Seq[RawRow] =
+  def rowsToRawItems(nonKeyColumnNames: Seq[String], rows: Seq[Row]): Seq[RawRow] =
     rows.map(
       row =>
         RawRow(
           Option(row.getString(row.fieldIndex(temporaryKeyName)))
             .getOrElse(throw new CdfSparkIllegalArgumentException("\"key\" can not be null.")),
-          io.circe.parser
-            .decode[Map[String, Json]](
-              mapper.writeValueAsString(row.getValuesMap[Any](nonKeyColumnNames)))
-            .right
-            .get
+          nonKeyColumnNames.map(f => f -> anyToRawJson(row.getAs[Any](f))).toMap
       ))
 
+  // scalastyle:off cyclomatic.complexity
+  /** Maps the object from Spark into circe Json object. This is the types we need to cover:
```

Comment:
Where did you get the list below from? 😅

Reply:
It's copy paste from documentation of `Row.get(...)` function